plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
+                        .addCopies(new FileOperation(MANIFEST, OUTPUT_FILE_CREATE_OR_OVERWRITE), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, OUTPUT_FILE_CREATE_OR_OVERWRITE), 1)
+                        .addCopies(new FileOperation(METADATA, OUTPUT_FILE_LOCATION), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, OUTPUT_FILE_LOCATION), 1)
+                        .build());
+    }
+
+    @Test
+    public void testSelect()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_select AS SELECT 1 col_name", 1);
+        assertFileSystemAccesses("SELECT * FROM test_select",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
+    }
+
+    @Test
+    public void testSelectWithFilter()
+            throws Exception
     {
-        return getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query).getQueryId().getId();
+        assertUpdate("CREATE TABLE test_select_with_filter AS SELECT 1 col_name", 1);
+        assertFileSystemAccesses("SELECT * FROM test_select_with_filter WHERE col_name = 1",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
     }
 
-    private static OperationCounts getCounts()
+    @Test
+    public void testJoin()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_join_t1 AS SELECT 2 as age, 'id1' AS id", 1);
+        assertUpdate("CREATE TABLE test_join_t2 AS SELECT 'name1' as name, 'id1' AS id", 1);
+
+        assertFileSystemAccesses("SELECT name, age FROM test_join_t1 JOIN test_join_t2 ON test_join_t2.id = test_join_t1.id",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 6)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 6)
```

Comment:
is it same file 6 times, or different files?

(same for INPUT_FILE_GET_LENGTH above)

Reply:
for both INPUT_FILE_STREAM and INPUT_FILE_GET_LENGTH those 6 comes from operations on 2 files - 1 file per table. 
I didn't find a good way to distinguish between them. 
- FileId from TrackingFileIoProvider doesnt help here (as it is actually different for each of them even tough these are the same files). 
- It is also impossible to predict these files names. 
- I was thinking about incorporating tableName into FileOperation and use it to distinguish between those files but that looked more complex 

In the end I decided that this is fine for now.